ADR/0035-artifact-streams.md

```diff
@@ -0,0 +1,150 @@
+
+Along with the [gitops-service] which was decided to stop deploying in [ADR 32. Decoupling Deployment],
+the controllers' APIs and the UI for AppStudio have been tightly coupled with the concept of [Application]s
+and [Component]s. For example, 
+
+- PipelineRuns triggered by PipelinesAsCode produced a single image that is tied to a specific [Component]
+  which is assumed to be an OCI artifact.
+- The AppStudio UI (HAC) display PipelineRuns for all [Component]s in a single [Application] together.
+- [Component]s are tied to artifacts produced from a single a git repository and branch.
+
+This model presented multiple challenges including:
+
+- If a user wants to build a new branch of an already onboarded git repository, they would need to create
+  a new [Component] (with a workspace-unique name).
+- [Component]s can be members of only one [Application].
+- As the number of related [Component]s increases, it becomes much harder to visualize all of the currently
+  or previously running PipelineRuns since all [Component]s are included together. This has resulted in users
+  creating multiple [Application]s that are related to each other just to resolve the visual representation
+  challenges.
+- Since a PipelineRun produces a single [Component], it becomes harder to support build strategies that produce
+  multiple artifacts, i.e [ko].
+- [Component]s are only defined in the context of an [Application] and cannot be moved. When a user wants to
+  start building a Component, they need to either add it to a current [Application] and have its builds affecting
+  the IntegrationTestScenarios results of the generated Snapshots or the component needs to be created in its
+  own temporary   Application only to be deleted and recreated in the appropriate one once the builds are
+  properly functioning.
+
+## Decision
+
+We are going to abstract the concepts of the built artifacts and their relationships to each other outside of
+any other controller within AppStudio. If a controller wants to maintain complete separation from others (i.e.
+in order to be fully deployable on its own), it may use an adapter pattern to convert this abstraction to its
+own relevant internal data model.
+
+### ArtifactConfig
+
+An `ArtifactConfig` describes the metadata which is used to define an `Artifact`, an `ArtifactStream`, and
+how `Artifact`s produced update that stream.
+
+`Artifact`s do not exist in isolation to other `Artifact`s in the system, however, there are both explicit
+and implicit relationships that exist between them, for example:
+
+- (implicit) Multiple `ArtifactConfig`s can be created for the same repository and branch combinations. These
+  configurations would be allowed as long as the name of the produced artifact is unique. This supports use
+  cases like `ko` where multiple `Artifact`s would be produced from a single build
+- (explicit) A single `ArtifactConfig` can describe the configuration for producing the "same" `Artifact"
```

Comment:
What is implied by putting "same" in quotes? Do you just mean that two different commits with two different git digests would produces two different binary images, but that two different ArtifactConfigs could be used to pipe them to a single ArtifactStream? Since they both feed the same ArtifactStream they can be thought of as being "the same" in that way?

Reply:
This use case would be to support versioning which is tied to the branches. There is likely little to no difference in how the `Artifact` is build from one branch to the next, but the content of the artifact changes.